src/unix/darwin.c

```diff
@@ -180,11 +185,144 @@ int uv_uptime(double* uptime) {
+      goto out;                                                               \
+  }                                                                           \
+  while (0)
+  V(iokit_handle, IOMasterPort);
+  V(iokit_handle, IOServiceMatching);
+  V(iokit_handle, IOServiceGetMatchingServices);
+  V(iokit_handle, IOIteratorNext);
+  V(iokit_handle, IOObjectRelease);
+  V(iokit_handle, IORegistryEntryCreateCFProperty);
+  V(core_foundation_handle, CFStringCreateWithCString);
+  V(core_foundation_handle, CFStringGetSystemEncoding);
+  V(core_foundation_handle, CFDataGetBytePtr);
+  V(core_foundation_handle, CFDataGetLength);
+  V(core_foundation_handle, CFDataGetBytes);
+  V(core_foundation_handle, CFRelease);
+#undef V
+
+#define S(s) pCFStringCreateWithCString(NULL, (s), kCFStringEncodingUTF8)
+  io_iterator_t it;
+  io_object_t service;
+
+  kr = pIOMasterPort(MACH_PORT_NULL, &mach_port);
+  // TODO(evanlucas) handle non KERN_SUCCESS return values
+  CFMutableDictionaryRef classes_to_match
+      = pIOServiceMatching("IOPlatformDevice");
+  kr = pIOServiceGetMatchingServices(mach_port, classes_to_match, &it);
+  // TODO(evanlucas) handle non KERN_SUCCESS return values
+  service = pIOIteratorNext(it);
+
+  CFStringRef device_type_str = S("device_type");
+  CFStringRef clock_frequency_str = S("clock-frequency");
+
+  do {
+    if (service) {
+      CFDataRef data = NULL;
+      data = pIORegistryEntryCreateCFProperty(service,
+                                              device_type_str,
+                                              NULL,
+                                              0);
+      if (data) {
+        const UInt8 *raw = pCFDataGetBytePtr(data);
+        if (strncmp((char *)raw, "cpu", 3) == 0
+          || strncmp((char *)raw, "processor", 9) == 0) {
+
+          data = pIORegistryEntryCreateCFProperty(service,
+                                                  clock_frequency_str,
+                                                  NULL,
+                                                  0);
+          if (data) {
+            uint32_t freq;
+            CFIndex len = pCFDataGetLength(data);
+            CFRange range;
+            range.location = 0;
+            range.length = len;
+
+            pCFDataGetBytes(data, range, (UInt8 *)&freq);
+            *speed = freq;
+            pCFRelease(data);
+            break;
+          }
+        }
+        pCFRelease(data);
+      }
+    }
+    service = pIOIteratorNext(it);
+  } while (service != 0);
+
+  pIOObjectRelease(it);
+
+  err = 0;
+out:
+  if (core_foundation_handle != NULL)
+    dlclose(core_foundation_handle);
+
+  if (iokit_handle != NULL)
+    dlclose(iokit_handle);
+
+  mach_port_deallocate(mach_task_self(), mach_port);
+
+  return err;
+}
+
 int uv_cpu_info(uv_cpu_info_t** cpu_infos, int* count) {
+  uint64_t cpuspeed = 0;
+  if (uv__get_cpu_speed(&cpuspeed))
+    return UV__ERR(errno);
```

Comment:
1. Zeroing `cpuspeed` isn't necessary, AFAICT.

2. The if block should be moved after the declarations.

3. `return UV__ERR(errno)` looks wrong because I don't think `uv__get_cpu_speed()` always sets `errno`. You probably want something like this?
```c
err = uv__get_cpu_speed(&cpuspeed);
if (err < 0)
  return err;
```

Reply:
thanks, updated